### ISSUE TITLE
Go workspace support from Go 1.18

### ISSUE THREAD

**Issue by gnuletik** [ldez/gomoddirectives#3]
Here is a repo to reproduce

```bahs
git clone https://github.com/gnuletik/gomoddirectives-repro
cd gomoddirectives-repro/
bash -c 'cd hello && golangci-lint cache clean && golangci-lint run --config ../.golangci-lint.yml'
```

It should show a warning:
```
WARN [linters context] running gomoddirectives failed: failed to get module file: unmarshaling error: invalid character '{' after top-level value: {
	"Path": "example.com/hello",
	"Main": true,
	"Dir": "/tmp/gomoddirectives-repro/hello",
	"GoMod": "/tmp/gomoddirectives-repro/hello/go.mod",
	"GoVersion": "1.18"
}
{
	"Path": "example.com/pkg",
	"Main": true,
	"Dir": "/tmp/gomoddirectives-repro/pkg",
	"GoMod": "/tmp/gomoddirectives-repro/pkg/go.mod",
	"GoVersion": "1.18"
}
: if you are not using go modules it is suggested to disable this linter
```

see https://github.com/golangci/golangci-lint/issues/2752

**Comment by ldez** [comment]
Hello,

can you run the following command: `go list -m -json`

**Comment by gnuletik** [comment]
Thanks for the fast feedback!

There you go
```
$ go list -m -json
{
	"Path": "example.com/hello",
	"Main": true,
	"Dir": "/tmp/gomoddirectives-repro/hello",
	"GoMod": "/tmp/gomoddirectives-repro/hello/go.mod",
	"GoVersion": "1.18"
}
{
	"Path": "example.com/pkg",
	"Main": true,
	"Dir": "/tmp/gomoddirectives-repro/pkg",
	"GoMod": "/tmp/gomoddirectives-repro/pkg/go.mod",
	"GoVersion": "1.18"
}
```

**Comment by ldez** [comment]
can you do a `go mod tidy` in the `hello` repo?

**Comment by gnuletik** [comment]
No I can't. It doesn't work because the modules are not published on fetchable URL, see https://github.com/golang/go/issues/50750

```
$ go mod tidy
go: finding module for package example.com/pkg
example.com/hello imports
	example.com/pkg: cannot find module providing package example.com/pkg: unrecognized import path "example.com/pkg": reading https://example.com/pkg?go-get=1: 404 Not Found
```

**Comment by ldez** [comment]
Fixed by 3ce9a6500b555d88b44e6f7ef7dafe4d88c7084b

**Comment by gnuletik** [comment]
Works like a charm, thanks :)
